internal/presentation/currencyRate.go

```diff
@@ -0,0 +1,28 @@
+package presentation
+
+import (
+	"github.com/gofiber/fiber/v2"
+	"go_service/internal/common"
+)
+
+type CurrencyHandlers struct {
+	currencyGateway common.CurrencyReader
+}
+
+func InitCurrencyHandlers(currencyGateway common.CurrencyReader) CurrencyHandlers {
```

Comment:
as for me 
NewCurrencyHandlers is better for constructor. Also, you use Init... Get, lest use one name for constructors.


Also, its better to describe interface in this file. 


```go

type CurrencyReader interface {
	GetUSDCurrencyRate() (float32, error)
}

type CurrencyHandlers struct {
	currencyGateway CurrencyReader
}

func InitCurrencyHandlers(currencyGateway CurrencyReader) CurrencyHandlers {
	return CurrencyHandlers{currencyGateway}
}

func (ch *CurrencyHandlers) GetCurrency(c *fiber.Ctx) error {
	rate, err := ch.currencyGateway.GetUSDCurrencyRate()

	if err != nil {
		return fiber.ErrInternalServerError
	}

	response := map[string]interface{}{
		"rate": rate,
	}

	return c.JSON(response)
}```